pinot-server/src/main/java/org/apache/pinot/server/starter/helix/IngestionBasedConsumptionStatusChecker.java

```diff
@@ -77,6 +77,9 @@ public int getNumConsumingSegmentsNotReachedIngestionCriteria() {
         LLRealtimeSegmentDataManager rtSegmentDataManager = (LLRealtimeSegmentDataManager) segmentDataManager;
         if (isSegmentCaughtUp(segName, rtSegmentDataManager)) {
           _caughtUpSegments.add(segName);
+          rtSegmentDataManager.notifyConsumptionCaughtUp(false);
```

Comment:
I believe there is a problem here. 

The catching up that happens in this phase is only when a server starts up, not for every consuming segment that is created when a server is already up and running.

My suggestion is to NOT wait for the server to catch up in order to report delays.

if we _do_ want to wait, then this wait should be at a table level.

Reply:
> only when a server starts up

This was the scenario we wanted to handle because often there is lot of catch-up to do after restarting servers. Not for every consuming segment on an already running server. 

>if we do want to wait, then this wait should be at a table level.

What is your rationale behind this?

Reply:
Code now changed to use server ready signal